### PR TITLE
[TR] PIM-4928 : media migration 1.3 - 1.4

### PR DESCRIPTION
**Pull request by jmleroux**
| Q                 | A
| ----------------- | ---
| Added Specs       |n
| Added Behats      |n
| Travis CI is ok   |y
| Changelog updated |n

### CODE REVIEW

**Comment by sumbobyboys** [comment]
Could you please explain those PR ? I don't understand them

**Comment by jmleroux** [comment]
@sumbobyboys , the workflow of this PR could resumed like this : 
- move all files under 'upload/product' to the new flysystem mount point
- update the old files references in product values
- update the old files references in product templates
- update the old file reference in product draft (EE only)
- update the old file reference in published product (EE only)

**Comment by sumbobyboys** [comment]
In this case, we should rework it because these steps seems simple and whan you read the code it's not obvious at all :/

**Comment by jjanvier** [comment]
this is the same thing that existed before except JM splitted ORM/Mongo classes in order to be more readable

he also handled the migrations for variant groups and drafts

**Comment by sumbobyboys** [comment]
Ok, then, I don't find it really readable, maybe not possible to do better. It was just a simple observation :)

File: upgrades/1.3-1.4/mongodb/migrate_medias.php [hunk 1]
```diff
@@ -27,7 +29,9 @@
 
 $migration->createFileInfoTable();
 $migration->storeLocalMedias();
-$migration->setOriginalFilenameToMedias($valueTable, null);
-$migration->migrateMediasOnProductValue($valueTable, null, null);
+
+$migration->setOriginalFilenameToMedias($valueTable);
+$migration->migrateMediasOnProductValue($valueTable);
+
```
Comment (jjanvier) [inline upgrades/1.3-1.4/mongodb/migrate_medias.php]: misssing the migration of product template

**Comment by jjanvier** [comment]
GTM

**Comment by nidup** [comment]
Not easy to review but GTM :)

**Pull request merged**